packages/interactivity/src/proxies/signals.ts

```diff
@@ -82,6 +87,30 @@ export class PropSignal {
 		this.update( { get: getter } );
 	}
 
+	/**
+	 * Changes the internal getter asynchronously.
+	 *
+	 * The update is made in a microtask, which prevents issues with getters
+	 * accessing the state, and ensures the update occurs before any render.
+	 *
+	 * @param getter New getter.
+	 */
+	public setGetterAsync( getter: () => any ) {
+		this.pendingGetter = getter;
+		Promise.resolve().then( () => this.consolidateGetter() );
```

Comment:
There is a complete overlap between browsers that support import maps and browsers that support `queueMicrotask`, so `queueMicrotask` can be used directly.

```suggestion
		queueMicrotask( () => this.consolidateGetter() );
```

Reply:
Great to know. 😄  I wasn't 100% sure, so I stayed with the old approach. I'll update this.